NDTensors/src/lib/GradedAxes/src/fusion.jl

```diff
@@ -6,6 +6,8 @@ OneToOne() = OneToOne{Bool}()
 Base.first(a::OneToOne) = one(eltype(a))
 Base.last(a::OneToOne) = one(eltype(a))
 
+gradedisequal(::OneToOne, ::OneToOne) = true
```

Comment:
It seems a bit strange to have this as a special definition.

I would imagine instead defining:
```julia
gradedisequal(r1::AbstractUnitRange, r2::AbstractUnitRange) = r1 == r2
```
(or maybe defined in terms of `blockisequal`, which should have a fallback definition for non-blocked unit ranges) and then defining equality for `OneToOne`.

Reply:
Another way of looking at this would be to make it so the generic definition:
```julia
blockisequal(a1, a2) && (blocklabels(a1) == blocklabels(a2))
```
"just works" for ranges of type `OneToOne`, which means that `blockisequal` and `blocklabels` would need to be defined for `OneToOne`. Maybe `blocklabels` doesn't make sense for non-graded unit ranges, so then we should have two definitions:
```julia
function gradedisequal(a1::AbstractUnitRange, a2::AbstractUnitRange)
  return blockisequal(a1, a2)
end

function gradedisequal(a1::AbstractGradedUnitRange, a2::AbstractGradedUnitRange)
  return blockisequal(a1, a2) && (blocklabels(a1) == blocklabels(a2))
end
```
and then ranges like `OneToOne` just need to make sure that `blockisequal` works correctly, which probably just requires implementing some part of the [block axis interface](https://juliaarrays.github.io/BlockArrays.jl/stable/man/abstractblockarrayinterface).

Reply:
I plan to deal with this in #1531 (see https://github.com/ITensor/ITensors.jl/pull/1531/commits/14c95cac9c7ed421ed6b05f8c723e46fdf59bad9 that already implements the suggestion)

Concerning `gradedisequal(::OneToOne, ::OneToOne) = true`, the aim is to provide a static result at compile time that can be propagated. I can remove the line if you prefer.

Reply:
I would prefer to have as few specialized definitions as possible for `OneToOne`, I doubt those kinds of definitions are critical for performance anywhere and we can always add them as needed.